chihaya.go

```diff
@@ -12,10 +12,50 @@ import (
 )
 
 // PeerID represents a peer ID.
-type PeerID string
+type PeerID [20]byte
 
-// InfoHash represents an infohash in hexadecimal notation.
-type InfoHash string
+func PeerIDFromBytes(b []byte) PeerID {
+	if len(b) != 20 {
+		panic("peer ID must be 20 bytes")
+	}
+
+	a := [20]byte{}
+	copy(a[:], b)
+	return PeerID(a)
+}
+
+func PeerIDFromString(s string) PeerID {
+	if len(s) != 20{
+		panic("peer ID must be 20 bytes")
+	}
+
+	a := [20]byte{}
```

Comment:
Let's replace this line with `var buf [20]byte`.
For initializing a zero value, var is usually preferred.